src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/CompositorWindow.cs

```diff
@@ -304,6 +309,182 @@ private void RecordingGUI()
+                bool running = compositionManager != null && compositionManager.TextureManager != null;
+                if (running)
+                {
+                    if (!compositionManager.IsOcclusionSettingSupported(compositionManager.OcclusionMode))
+                    {
+                        RenderTitle("The current camera does not support occlusion.", Color.clear);
+                    }
+                    else if (compositionManager.OcclusionMode == OcclusionSetting.BodyTracking)
+                    {
+                        RenderTitle("Body Tracking Settings", Color.clear);
+                        EditorGUILayout.BeginHorizontal();
+                        {
+                            GUIContent label = new GUIContent("Blur Size");
+                            compositionManager.TextureManager.blurSize = EditorGUILayout.Slider(
+                                label,
+                                compositionManager.TextureManager.blurSize, 0, 10);
+                        }
+                        EditorGUILayout.EndHorizontal();
+                        EditorGUILayout.BeginHorizontal();
+                        {
+                            GUIContent label = new GUIContent("Number of Blur Passes");
+                            compositionManager.TextureManager.numBlurPasses = (int) EditorGUILayout.Slider(
+                                label,
+                                compositionManager.TextureManager.numBlurPasses, 1, 5);
+                        }
+                        EditorGUILayout.EndHorizontal();
+                    }
+                    else if (compositionManager.OcclusionMode == OcclusionSetting.RawDepthCamera)
+                    {
+                        RenderTitle("Raw Depth Camera Settings", Color.clear);
+                        EditorGUILayout.BeginHorizontal();
+                        {
+                            GUIContent label = new GUIContent("Blur Size");
+                            compositionManager.TextureManager.blurSize = EditorGUILayout.Slider(
+                                label,
+                                compositionManager.TextureManager.blurSize, 0, 10);
+                        }
+                        EditorGUILayout.EndHorizontal();
+                        EditorGUILayout.BeginHorizontal();
+                        {
+                            GUIContent label = new GUIContent("Number of Blur Passes");
+                            compositionManager.TextureManager.numBlurPasses = (int) EditorGUILayout.Slider(
+                                label,
+                                compositionManager.TextureManager.numBlurPasses, 1, 5);
+                        }
+                        EditorGUILayout.EndHorizontal();
+                    }
+                }
+                else
+                {
+                    RenderTitle("Updating occlusion settings is not possible when the compositor isn't running.", Color.green);
```

Comment:
>RenderTitle("Updating occlusion settings is not possible when the compositor isn't running.", Color.green); [](start = 20, length = 107)

Seems strange to me that you can't update the settings when the compositor isn't running. Is this just because the settings are stored on the TextureManager? If that's the only reason, I'd propose storing the settings on the CompositionManager and letting the TextureManager read them from there.

Reply:
opened: https://github.com/microsoft/MixedReality-SpectatorView/issues/332